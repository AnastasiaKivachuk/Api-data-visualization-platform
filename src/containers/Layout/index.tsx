import type {NextPage} from 'next'
import React from "react";
import styled from "@emotion/styled";

import Grid from '@mui/material/Grid'
import {Container} from "@mui/material";

import Menu from "../../components/Menu";

type Props = {
    children: JSX.Element
}

const Layout: NextPage<Props> = ({children}) => {
    const StyledContainer = styled(Container)`
      padding-top: 60px;
      display: flex;
      flex-direction: column;
      align-items: center;
    `
    return (
        <Grid item>
            <Menu/>
            <StyledContainer>{children}</StyledContainer>
        </Grid>
    )
}

export default Layout
