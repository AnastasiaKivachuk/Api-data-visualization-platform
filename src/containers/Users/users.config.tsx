import * as React from 'react';

import {GridColDef, GridValueGetterParams} from '@mui/x-data-grid';

import {getDateWithFormat} from "../../helpers/functions.helpers";

export const usersColumns: GridColDef[] = [
    {field: 'id', headerName: 'id', sortable: false, width: 200},
    {
        field: 'name', headerName: 'Name', width: 230,
        valueGetter: (params: GridValueGetterParams) => params.row.name || '-'
    },
    {
        field: 'rocket', headerName: 'Rocket', width: 230,
        valueGetter: (params: GridValueGetterParams) => params.row.rocket || '-'
    },
    {
        field: 'twitter',
        headerName: 'Twitter',
        valueGetter: (params: GridValueGetterParams) => params.row.twitter || '-',
        width: 230
    },
    {
        field: 'timestamp',
        headerName: 'Date registration',
        valueGetter: (params: GridValueGetterParams) => getDateWithFormat(params.row.timestamp),
        width: 230
    },
];