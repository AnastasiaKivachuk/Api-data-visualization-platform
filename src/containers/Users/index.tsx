import React from "react";

import Typography from "@mui/material/Typography";
import {DataGrid} from "@mui/x-data-grid";

import {usersColumns} from "./users.config";
import {UsersInterface} from "../../interfaces/users.interface";

const Users = ({users}: UsersInterface) => {
    return (
        <>
            <Typography variant="h2" textAlign='center'>Users</Typography>

            <div style={{height: 640, width: '100%'}}>
                {(users || [])?.length > 0 ?
                    <DataGrid
                        rows={users || []}
                        columns={usersColumns}
                        pageSize={10}
                        rowsPerPageOptions={[10]}
                    /> :
                    <Typography variant="h4" textAlign='center'>No Data</Typography>
                }
            </div>

        </>
    )
}

Users.defaultProps = {
    users: []
}

export default Users
