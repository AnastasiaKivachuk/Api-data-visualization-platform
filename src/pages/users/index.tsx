import type {NextPage} from 'next'
import React from "react";

import Users from "../../containers/Users";
import {UsersInterface} from "../../interfaces/users.interface";
import {getUsers} from "../../services/api/users.api";
import Custom404Page from "../404";


export async function getStaticProps() {
    try {
        const {data} = await getUsers();
        return {
            props: {
                users: data.users,
            },
        };
    } catch (err) {
        return {props: {isError: true}};
    }
}

interface Props extends UsersInterface {
    isError?: boolean
}

const UsersPage: NextPage<Props> = ({users, isError}) => {
    return (
        isError ? <Custom404Page/> : <Users users={users}/>
    )
}

export default UsersPage
