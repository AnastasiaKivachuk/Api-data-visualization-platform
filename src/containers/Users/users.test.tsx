import {shallow} from "enzyme";
import React from "react";

import Layout from "./index";
import Users from "./index";
import {UsersInterface} from "../../interfaces/users.interface";
import {usersColumns} from "./users.config";

const MockUsersTable = [{
    id: "577073b9-1aac-4952-a673-9d4cf1895c51",
    name: "defaultQ1213456789",
    rocket: null,
    timestamp: "2022-06-14T00:34:47.625834+00:00",
    twitter: null,
}]

const setUp = (props: UsersInterface | {}) => shallow(<Users {...props}/>);

describe("Users component", () => {

    it("should render Users component with default props", () => {
        const component = setUp({});
        expect(component).toMatchSnapshot();
    })

    it("should render Users component with props", () => {
        const component = setUp({users: MockUsersTable});
        expect(component).toMatchSnapshot();
    })

    it("hide table and show info if no props", () => {
        const component = setUp({});
        const text = ['Users', 'No Data'];
        expect(component.find('Memo(ForwardRef(DataGrid))').length).toBe(0)
        expect(component.find('ForwardRef(Typography)').length).toBe(2)
        component.find('ForwardRef(Typography)').map((item, index) => {
            expect(item.text()).toBe(text[index]);
        })
    })

    it("hide info and show table if props", () => {
        const component = setUp({users: MockUsersTable});
        expect(component.find('Memo(ForwardRef(DataGrid))').length).toBe(1)
        expect(component.find('ForwardRef(Typography)').length).toBe(1)
        expect(component.find('ForwardRef(Typography)').text()).toBe('Users')
    })

    it("check usersColumns config", () => {
        expect(usersColumns).toMatchSnapshot();
    })
})
