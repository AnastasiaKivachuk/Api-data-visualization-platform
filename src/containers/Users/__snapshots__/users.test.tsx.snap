// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component check usersColumns config 1`] = `
Array [
  Object {
    "field": "id",
    "headerName": "id",
    "sortable": false,
    "width": 200,
  },
  Object {
    "field": "name",
    "headerName": "Name",
    "valueGetter": [Function],
    "width": 230,
  },
  Object {
    "field": "rocket",
    "headerName": "Rocket",
    "valueGetter": [Function],
    "width": 230,
  },
  Object {
    "field": "twitter",
    "headerName": "Twitter",
    "valueGetter": [Function],
    "width": 230,
  },
  Object {
    "field": "timestamp",
    "headerName": "Date registration",
    "valueGetter": [Function],
    "width": 230,
  },
]
`;

exports[`Users component should render Users component with default props 1`] = `
<Fragment>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h2"
  >
    Users
  </ForwardRef(Typography)>
  <div
    style={
      Object {
        "height": 640,
        "width": "100%",
      }
    }
  >
    <ForwardRef(Typography)
      textAlign="center"
      variant="h4"
    >
      No Data
    </ForwardRef(Typography)>
  </div>
</Fragment>
`;

exports[`Users component should render Users component with props 1`] = `
<Fragment>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h2"
  >
    Users
  </ForwardRef(Typography)>
  <div
    style={
      Object {
        "height": 640,
        "width": "100%",
      }
    }
  >
    <Memo(ForwardRef(DataGrid))
      columns={
        Array [
          Object {
            "field": "id",
            "headerName": "id",
            "sortable": false,
            "width": 200,
          },
          Object {
            "field": "name",
            "headerName": "Name",
            "valueGetter": [Function],
            "width": 230,
          },
          Object {
            "field": "rocket",
            "headerName": "Rocket",
            "valueGetter": [Function],
            "width": 230,
          },
          Object {
            "field": "twitter",
            "headerName": "Twitter",
            "valueGetter": [Function],
            "width": 230,
          },
          Object {
            "field": "timestamp",
            "headerName": "Date registration",
            "valueGetter": [Function],
            "width": 230,
          },
        ]
      }
      pageSize={10}
      rows={
        Array [
          Object {
            "id": "577073b9-1aac-4952-a673-9d4cf1895c51",
            "name": "defaultQ1213456789",
            "rocket": null,
            "timestamp": "2022-06-14T00:34:47.625834+00:00",
            "twitter": null,
          },
        ]
      }
      rowsPerPageOptions={
        Array [
          10,
        ]
      }
    />
  </div>
</Fragment>
`;
